src/include/ngf/context.h

```diff
@@ -39,7 +39,7 @@ typedef void (*NContextValueChangeFunc) (NContext *context,
  *
  * @param context NContext structure.
  * @param key Key.
- * @param value Values as NValue type.
+ * @param value Values as NValue type. Context takes over the ownership.
```

Comment:
Does it take ownership? Always? Test code adds the same value multiple times. Couple of times with other args being invalid and then once with all valid args. Lack of double free hiccups -> only the last successful op takes ownership? This in a void function where success/not is a bit hard to know... I'd consdider making some more adjustments to docs/impl/tests

Reply:
Well, the funny thing is that it doesn't take the ownership on error case if context or key is NULL. And of course doesn't tell that it didn't succeed. The test used to assume it never takes ownership and still assumes that it only takes ownership in the case where parameters are valid.

Do agree that this isn't good